### PR TITLE
Personal/gankoji/refactor restriction messages

### PR DESCRIPTION
**Pull request by gankoji**
Refactor member restrictions messages. 

Previously, both restriction creation (Member banned) and restriction expiration (member unbanned) were sent as messages through the same pipe/topic in events service. To make this easier to sort out for other services, we've split those into two message streams. The pubsub, which makes its way to the front end, stays as one message stream with extra metadata as to which message type we're sending, but does need some refactoring for the type changes.

### CODE REVIEW

File: clients/web/src/lib/api/queries/RestrictedPlayers.store.ts [hunk 1]
```diff
@@ -110,12 +109,29 @@ export class RestrictedPlayersStore extends LiveQueryStore<RestrictedPlayersStor
...
+            const memberRestriction = {
+                id: message.data.followRestrictedMembers.id,
+                type: message.data.followRestrictedMembers.type,
+                expiration: message.data.followRestrictedMembers.expiration,
+                reason: message.data.followRestrictedMembers.reason,
+                member: message.data.followRestrictedMembers.member,
+                manualExpiration: message.data.followRestrictedMembers.manualExpiration,
+                manualExpirationReason: message.data.followRestrictedMembers.manualExpirationReason,
+                memberId: message.data.followRestrictedMembers.memberId,
+            };
```
Comment (bwpikaard) [inline clients/web/src/lib/api/queries/RestrictedPlayers.store.ts:127]:
```suggestion
            const {eventType, restriction} = message.data.followRestrictedMembers;
```

File: clients/web/src/lib/api/queries/RestrictedPlayers.store.ts [hunk 2]
```diff
@@ -20,32 +20,30 @@ interface Member {
...
+    manualExpirationReason?: string;
 
-        manualExpirationReason?: string;
+    memberId: number;
+}
 
-        memberId: number;
-    };
+export interface MemberRestrictionEvent extends MemberRestriction{
+    eventType: number;
```
Comment (bwpikaard) [inline clients/web/src/lib/api/queries/RestrictedPlayers.store.ts:42]: Should this be a number or enum for clarity?

**Review by bwpikaard — APPROVED** [review]

File: clients/web/src/lib/api/queries/RestrictedPlayers.store.ts [hunk 3]
```diff
@@ -110,12 +109,29 @@ export class RestrictedPlayersStore extends LiveQueryStore<RestrictedPlayersStor
...
+                type: message.data.followRestrictedMembers.type,
+                expiration: message.data.followRestrictedMembers.expiration,
+                reason: message.data.followRestrictedMembers.reason,
+                member: message.data.followRestrictedMembers.member,
+                manualExpiration: message.data.followRestrictedMembers.manualExpiration,
+                manualExpirationReason: message.data.followRestrictedMembers.manualExpirationReason,
+                memberId: message.data.followRestrictedMembers.memberId,
+            };
+
             switch (message.data.followRestrictedMembers.eventType) {
```
Comment (bwpikaard) [inline clients/web/src/lib/api/queries/RestrictedPlayers.store.ts:129]:
```suggestion
            switch (eventType) {
```

File: clients/web/src/lib/api/queries/RestrictedPlayers.store.ts (hunk 2 above)
Comment (bwpikaard) [reply clients/web/src/lib/api/queries/RestrictedPlayers.store.ts:42]: Actually, it's used as an enum in the switch case below?

Comment (gankoji) [reply clients/web/src/lib/api/queries/RestrictedPlayers.store.ts]: done

**Pull request merged**